File: searchInsert/searchInsert.go
package main

func searchInsert(nums []int, target int) int {
	var min, mid int
	max := len(nums) - 1

	for min <= max {
		mid = (min + max) / 2
		if target < nums[mid] {
			max = mid - 1
		} else if target > nums[mid] {
			min = mid + 1
		} else {
			return mid
		}
	}
	if nums[mid] <= target {
		return mid + 1
	} else {
		return mid
	}
}

File: java/easy/SearchInsert/SearchInsert.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int min, mid;
        int max = nums.length;
        min = mid = 0;

        while (min <= max) {
            mid = (min + max) / 2;
            if (target < nums[mid]) {
                max = mid - 1;
            } else if (target > nums[mid]) {
                min = mid + 1;
            } else {
                return mid;
            }
        }
        if (nums[mid] <= target) {
            return mid + 1;
        } else {
            return mid;
        }
    }
}